src/GitHub.App/Services/GistPublishService.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Reactive.Linq;
+using GitHub.Api;
+using GitHub.Models;
+using LibGit2Sharp;
+using Octokit;
+
+namespace GitHub.Services
+{
+    [Export(typeof(IGistPublishService))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public class GistPublishService : IGistPublishService
+    {
+        [ImportingConstructor]
+        public GistPublishService()
```

Comment:
FYI, with MEF, you don't need the `ImportingConstructor` attribute if there's an empty constructor (or none at all). You only need it if all constructors require parameters or if you don't want the empty constructor to be the one that gets called when MEF instantiates an instance.